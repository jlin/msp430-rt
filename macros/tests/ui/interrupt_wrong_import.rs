#![no_main]

// Incorrect w/ bad error message- use of undeclared crate or module `interrupt`.
use msp430_rt_macros::{entry, interrupt};

#[entry]
fn main() -> ! {
    unimplemented!()
}

#[interrupt]
fn TIM2() {
    unimplemented!()
}
