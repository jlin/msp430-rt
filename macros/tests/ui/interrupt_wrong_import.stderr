error[E0570]: "msp430-interrupt" is not a supported ABI for the current target
  --> tests/ui/interrupt_wrong_import.rs:11:1
   |
11 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: the extern "msp430-interrupt" ABI is experimental and subject to change
  --> tests/ui/interrupt_wrong_import.rs:11:1
   |
11 | #[interrupt]
   | ^^^^^^^^^^^^
   |
   = note: see issue #38487 <https://github.com/rust-lang/rust/issues/38487> for more information
   = help: add `#![feature(abi_msp430_interrupt)]` to the crate attributes to enable
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `interrupt`
  --> tests/ui/interrupt_wrong_import.rs:11:1
   |
11 | #[interrupt]
   | ^^^^^^^^^^^^ use of unresolved module or unlinked crate `interrupt`
   |
   = help: if you wanted to use a crate named `interrupt`, use `cargo add interrupt` to add it to your `Cargo.toml`
   = note: this error originates in the attribute macro `interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)
